x-pack/plugins/security_solution/public/management/components/privileged_route/privileged_route.tsx

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { ComponentType } from 'react';
+import React, { memo } from 'react';
+import { Route } from '@kbn/kibana-react-plugin/public';
+import { NoPrivilegesPage } from '../../../common/components/no_privileges';
+import { useIsExperimentalFeatureEnabled } from '../../../common/hooks/use_experimental_features';
+import { NoPermissions } from '../no_permissons';
+import { AdministrationSubTab } from '../../types';
+
+export interface PrivilegedRouteProps {
+  path: string;
+  component: ComponentType<{}>;
+  privilege: boolean;
+}
+
+export const PrivilegedRoute = memo(({ component, privilege, path }: PrivilegedRouteProps) => {
+  const isEndpointRbacEnabled = useIsExperimentalFeatureEnabled('endpointRbacEnabled');
+  const isEndpointRbacV1Enabled = useIsExperimentalFeatureEnabled('endpointRbacV1Enabled');
+
+  let componentToRender = component;
+
+  if (!privilege) {
+    const shouldUseMissingPrivilegesScreen =
+      isEndpointRbacEnabled ||
+      (isEndpointRbacV1Enabled && path.includes(AdministrationSubTab.responseActionsHistory));
```

Comment:
Why do we have route specific logic in this component?  Is it because Response Actions are the only ones that (in v1) are currently under RBAC control?

Also, not sure its the best idea to use `path.includes()` here.  Maybe just use this instead:

```typescript
path === MANAGEMENT_ROUTING_RESPONSE_ACTIONS_HISTORY_PATH
```

Reply:
Yes, it is because of this temporary situation where we need this page only for Response Actions History in v1. We'll be able to remove these lines after v2 is released.

Regarding `path.includes()`, I used it because I had some trouble with the full(er) path in the tests, but with a little more effort it is fixed now and more specific, thanks for the suggestion again!

04cde10